packages/puppeteer-core/src/cdp/Accessibility.ts

```diff
@@ -420,16 +406,10 @@ class AXNode {
         break;
     }
 
-    // Here and below: Android heuristics
-    if (this.#hasFocusableChild()) {
-      return false;
-    }
-    if (this.#focusable && this.#name && this.#name !== 'Document') {
```

Comment:
@Lightning00Blade I think we might have needed to keep `this.#focusable`  check as it might not be related to name checks